src/stores/repository-core/actions/repoMeta.ts

```diff
@@ -14,6 +16,21 @@ export const getRepositoryMetaFromContract = async (id: string): Promise<{ resul
   })
 }
 
+export const isRepositoryNameAvailable = async (name: string): Promise<boolean> => {
+  const userSigner = new InjectedArweaveSigner(window.arweaveWallet)
+  await userSigner.setPublicKey()
+  userSigner.getAddress = () => userSigner.signer.getActiveAddress()
+
+  const contract = getWarpContract(CONTRACT_TX_ID, userSigner)
```

Comment:
passing userSigner here for read function will charge users fee. we dont need signer for viewState calls.

Reply:
I removed usage of `userSigner` in this read function on the new commit by passing the caller itself.